### PR TITLE
Add initial content to the data sources interactive

### PR DESCRIPTION
**Pull request by Nguyen-Andy2**
**High level description:**

This PR adds the story content to the interactive data sources page. It includes different components for each scroller section in the navigation. It adds lottie animations to the page and includes the initial card copy that will be the overlay card for each animation segment. Note that some sections (like `Data Validation`) are still pending copy text for the cards (noted by "TBD") which may come later.

**Technical details:**

Each scroller section makes use of the `Scroller.jsx` component, `ScrollerOverlay.jsx` component, and `ScrollerOverlayCard.jsx` component to swap the overlay card with its respective backdrop on scroll. The underlying `react-scrollama` package relies on Intersection Observers. Animations in the backdrop use the `svg` renderer from `lottie-web`. They are triggered to play within the `onStepEnter` callback via the `playAnimation()` method.

**JIRA Ticket:**
As discussed with the developers, no JIRA ticket has been created for the interactive data sources feature.

**Mockup:**

The following are ALL required for the PR to be merged:

Author:
- [ ] Linked to this PR in JIRA ticket
- [ ] Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR `if applicable`
- [ ] Verified cross-browser compatibility: Chrome, Safari, Firefox, Edge
- [ ] Verified mobile/tablet/desktop/monitor responsiveness
- [ ] Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension)
- [ ] Added Unit Tests for helper functions, reducers, models and Container/Component Interactivity Expectations `if applicable` [React Testing Library](react-testing-library.md)
- [ ] [API contract](https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts) updated `if applicable`
- [ ] [Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated `if applicable`

Reviewer(s):
- [ ] Design review complete `if applicable`
- [ ] [API #1234](https://github.com/fedspendingtransparency/usaspending-api/pull/1234) merged concurrently `if applicable`
- [ ] Code review complete

### CODE REVIEW

**Comment by Nguyen-Andy2** [comment]
@anjenkin @tuckergordon Please note that there is still card text for the interactive sections that are still pending from the client, and this text will likely be provided after the Sep 19 handoff. For example, there is a `Data Validation` section that still has placeholder cards with `TBD` text and `Example Validation 1, 2, etc.`, and the client mentioned that a colleague was still working on providing that text for the interactive. However, given our time constraints on this work, we wanted to trigger the PR as soon as we had a content structure in place that would not require major revisions to the overall scroller/animation structure.

For the most part, any card copy still pending from the client is primarily for sections that have a more generic animation looping in the backdrop, where additional cards could be added/removed without needing to modify the animation itself. In contrast, cards in other sections that directly tie in to a specific animation in the backdrop (e.g. a card for `File A` that appears on screen when an animation specific to `File A` animates in the backdrop) would essentially be final, aside from any _minor_ text updates to existing cards that may be requested in the future. Any updates to those sections outside of minor text updates (for example, like adding brand new cards) would likely require future updates to the animation itself. After Sep 19, I think the majority of text changes that may occur would be ones that avoided needing changes to the animation.

Note: There was one request from the client late Friday to change the order of FPDS and FABS, as well as SAM and FSRS, in the `Award Data` section animation. Chris is working on updating that animation now, and then we would change the ordering of the overlay cards to match the new animation, but that is the only change to the animations expected today.

File: src/_scss/pages/interactiveDataSources/_aboutSection.scss [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    }
+
+    .interactives-guide__cardLine {
+        @media (max-width: $medium-screen) {
+            margin: 2px 16px 20px;
+            width: 100%;
+        }
+        width: 38%;
+        margin: 5px 0px 10px;
+        border-bottom: 3px solid #3b78bd;
```
Comment (anjenkin) [inline src/_scss/pages/interactiveDataSources/_aboutSection.scss:75]: Could a color variable be used here?

File: src/_scss/pages/interactiveDataSources/_aboutSection.scss [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+            width: 100%;
+        }
+        width: 38%;
+        margin: 5px 0px 10px;
+        border-bottom: 3px solid $color-primary;
+    }
+
+
+    .interactives-guide__background-green {
+        background: linear-gradient(to bottom, white 25%, #2a677b 25%);
```
Comment (anjenkin) [inline src/_scss/pages/interactiveDataSources/_aboutSection.scss:80]: color variable?

File: src/_scss/pages/interactiveDataSources/_aboutSection.scss [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    }
+
+    .interactives-guide_begin-scroller {
+        display: flex;
+        flex-direction: column;
+        align-items: center;
+        padding-bottom: 5%;
+
+        h4 {
+            color: #fff;
```
Comment (anjenkin) [inline src/_scss/pages/interactiveDataSources/_aboutSection.scss:94]: color variable?

File: src/_scss/pages/interactiveDataSources/_scrollerOverlayCard.scss [hunk 4]
```diff
@@ -68,5 +77,24 @@
...
+
+    
+    .glossary-term {
+      font-weight: $font-semibold;
+      color: $color-white;
+    }
+    .usda-glossary-link {
+      color: $color-white;
+      &:hover, &:active, &:focus, &:visited {
+          color: $color-gray-light;;
```
Comment (anjenkin) [inline src/_scss/pages/interactiveDataSources/_scrollerOverlayCard.scss:89]: double semicolon here

**Comment by anjenkin** [comment]
@Nguyen-Andy2 I noticed that when using the left side nav to switch between sections, the UI scrolls through either the current or next section overlays (depending on the sections you're scrolling to).  The scrolling overlays is a bit distracting because it's so fast you can't read the copy in the overlays.  Is this expected behavior?


https://user-images.githubusercontent.com/121010/191071158-388c9306-2b22-43f5-9898-0ba3ed3a13bf.mov

**Comment by Nguyen-Andy2** [comment]
> @Nguyen-Andy2 I noticed that when using the left side nav to switch between sections, the UI scrolls through either the current or next section overlays (depending on the sections you're scrolling to). The scrolling overlays is a bit distracting because it's so fast you can't read the copy in the overlays. Is this expected behavior?
> 
>  Screen.Recording.2022-09-19.at.12.35.49.PM.mov

@anjenkin Yes, that is expected. It is not intended to be able to read the copy in the overlays after clicking on a section nav item during the transition. It is used just as the transition to that section. This sidenav functionality was replicated from the latest sidenav created from the Agency Profiles section: https://www.usaspending.gov/agency/department-of-the-treasury?fy=2022

**Comment by anjenkin** [comment]
@Nguyen-Andy2 Also there are a lot of new accessibility issues introduced on this page.  All seem to pertain to color contrast issues and aria-labels on <g> elements.  I think the aria-labels issue are coming from the lottie library.  I'm more concerned about the color contrast issues.  Were you able to look into these at all?  Are these issue benign? 
Screen Shot 2022-09-19 at 12 59 48 PM

**Comment by Nguyen-Andy2** [comment]
> @Nguyen-Andy2 Also there are a lot of new accessibility issues introduced on this page. All seem to pertain to color contrast issues and aria-labels on elements. I think the aria-labels issue are coming from the lottie library. I'm more concerned about the color contrast issues. Were you able to look into these at all? Are these issues benign? Screen Shot 2022-09-19 at 12 59 48 PM

@ValGreens561 and I looked into these. Yeah those aria label issues seem to come from the Lottie library SVGs that are created. 

But for the contrast issues, it looks like many of those that were flagged by Axe were for the text/background contrast in the intro section, history section, and scroller overlay cards where the text is white and the background is not white. We reevaluated these color contrasts using https://webaim.org/resources/contrastchecker/ (plugging in the background color used compared to the overlay text color used) and it looks like on the website they are passing the minimum threshold that is described in the Axe test. I've attached some screenshots below :
image (5)
Screen Shot 2022-09-19 at 2 35 23 PM
Screen Shot 2022-09-19 at 2 43 59 PM

We also ran the WAVE extension tool on the page and it did not automatically detect any contrast issues:
image


Also, Axe raised a couple issues that originate from some shared components (such as the Glossary Link) that predate the changes introduced in this PR. For the glossary link, we've used that shared component in many of the scroller overlay cards.
Screen Shot 2022-09-19 at 2 56 25 PM

**Comment by Nguyen-Andy2** [comment]
^ Continuing on from the previous comment regarding the contrast issues raised, here is a screenshot of where Axe is reporting a contrast issue for the Appropriations card h4 tag. The background color it reports is correct, but the foreground color it uses (`#4d76a9`) is not the color we are using in the card for the tag (we are using `$color-white` from the `_variables.scss` file)  
image

**Review by anjenkin — APPROVED** [review]

**Pull request merged**